src/Evolution/Systems/Cce/AnalyticSolutions/SphericalMetricData.cpp

```diff
@@ -0,0 +1,385 @@
+               intermediate_dr_jacobian.get(j, 0)) *
+              get<1, 1>(intermediate_spherical_metric);
+      for (size_t k = 1; k < 3; ++k) {
+        intermediate_dr_cartesian_metric.get(i + 1, j + 1) +=
+            intermediate_jacobian.get(i, 0) * intermediate_jacobian.get(j, k) *
+                intermediate_dr_spherical_metric.get(1, k + 1) +
+            (intermediate_dr_jacobian.get(i, 0) *
+                 intermediate_jacobian.get(j, k) +
+             intermediate_jacobian.get(i, 0) *
+                 intermediate_dr_jacobian.get(j, k)) *
+                intermediate_spherical_metric.get(1, k + 1);
+
+        for (size_t l = 1; l < 3; ++l) {
+          intermediate_dr_cartesian_metric.get(i + 1, j + 1) +=
+              intermediate_jacobian.get(i, k) *
+                  intermediate_jacobian.get(j, l) *
+                  intermediate_dr_spherical_metric.get(k + 1, l + 1) +
+              (intermediate_dr_jacobian.get(i, k) *
+                   intermediate_jacobian.get(j, l) +
+               intermediate_jacobian.get(i, k) *
+                   intermediate_dr_jacobian.get(j, l)) *
+                  intermediate_spherical_metric.get(k + 1, l + 1);
+        }
+      }
+    }
+  }
+
+  Variables<
+      tmpl::list<::Tags::SpinWeighted<::Tags::TempScalar<0, ComplexDataVector>,
+                                      std::integral_constant<int, 0>>,
+                 ::Tags::SpinWeighted<::Tags::TempScalar<0, ComplexDataVector>,
+                                      std::integral_constant<int, 1>>,
+                 ::Tags::SpinWeighted<::Tags::TempScalar<2, ComplexDataVector>,
+                                      std::integral_constant<int, 0>>>>
+      buffers{Spectral::Swsh::number_of_swsh_collocation_points(l_max_)};
+
+  auto& derivative_buffer =
+      get(get<::Tags::SpinWeighted<::Tags::TempScalar<0, ComplexDataVector>,
+                                   std::integral_constant<int, 1>>>(buffers));
+  auto& pre_derivative_buffer =
+      get(get<::Tags::SpinWeighted<::Tags::TempScalar<0, ComplexDataVector>,
+                                   std::integral_constant<int, 0>>>(buffers));
+
+  for (size_t a = 0; a < 4; ++a) {
+    for (size_t b = a; b < 4; ++b) {
+      pre_derivative_buffer.data() = std::complex<double>(1.0, 0.0) *
+                                     intermediate_cartesian_metric.get(a, b);
+      Spectral::Swsh::angular_derivatives<
+          tmpl::list<Spectral::Swsh::Tags::Eth>>(
+          l_max_, 1, make_not_null(&derivative_buffer), pre_derivative_buffer);
+      for (size_t i = 0; i < 3; ++i) {
+        intermediate_d_cartesian_metric.get(i, a, b) =
+            intermediate_jacobian.get(i, 0) *
+                intermediate_dr_cartesian_metric.get(a, b) -
+            intermediate_jacobian.get(i, 1) * real(derivative_buffer.data()) -
+            intermediate_jacobian.get(i, 2) * imag(derivative_buffer.data());
+      }
+    }
+  }
+
+  // resample each component.
```

Comment:
and if you keep the resampling, can't it call a generic function to resample each component?